src/core_plugins/kibana/public/visualize/editor/styles/_editor.less

```diff
@@ -14,13 +14,37 @@
     height: 100%;
   }
 
+  @media (max-width: @screen-sm-max) {
+    visualization {
+      // While we are on a small screen the visualization is below the
+      // editor. In this cases it needs a minimum height, since it would otherwise
+      // maybe end up with 0 height since it just gets the flexbox rest of the screen.
+      min-height: 400px;
+    }
+  }
+
   .collapsible-sidebar {
     .flex-parent(0, 0, auto);
     margin-right: 10px;
     flex-direction: row;
-    max-width: 75%;
     min-width: @vis-editor-sidebar-min-width;
+    max-width: 100%;
     width: @vis-editor-sidebar-min-width;
+
+    @media (max-width: @screen-sm-max) {
+      // If we are on a small screen we force the editor to take 100% width.
+      // We cannot use a min-width: @screen-md-min query here, since we
+      // would otherwise break the manual resizing on large screens.
```

Comment:
hmm ? this styles only apply on small screens, so how would that break anything on large screens ?

Reply:
We usually try to use `min-width` media queries over `max-width`. So using a `min-width` query here would mean, we place the `width: 100%` outside any media query, but we would need to reset it then inside the media query (for medium/large screens). To reset it we could use something like `width: auto !important`, but that would now have priority over the `width` style that the rescaler actually sets to the element directly, and rescaling wouldn't work anymore. Thus we need to use `max-width` media query here, to limit the style to only small screens, and cannot work as usual: set it without media query, and change it on medium/large screens.